### DISCUSSION TITLE
APP layer hello_world does not work

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#589] [Q&A] [answered]
Hi, I'm trying to test the APP layer of the firmware. I did:
1. `git clone --recursive https://github.com/bitcraze/crazyflie-firmware.git`;
2. installed toolbelt `docker run --rm -it bitcraze/toolbelt`, and update;
3. build the app layer `tb make_app examples/app_hello_world/`;
4. build the firmware `tb make`;

it throws an error ```
/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld: src/built-in.o: in function `appTask':
/module/build/../src/modules/src/app_handler.c:59: undefined reference to `appMain'
collect2: error: ld returned 1 exit status
make[1]: *** [tools/kbuild/Makefile.kbuild:570: firmware] Error 1
make[1]: Leaving directory '/module/build'
make: *** [tools/kbuild/Makefile.kbuild:150: sub-make] Error 2
make: Leaving directory '/module'
Error: The call 'docker run --rm -u 1002 -v /var/run/docker.sock:/var/run/docker.sock -it -v /home/ingenious/Documents/drone/crazyflie-firmware:/module bitcraze/builder tools/build/make' failed with error code 2```

Do I need to modify the `oot.mk` file in the `/tools/make` folder? I'm sorry but the documentation on your website makes me confused.

**Comment by noahzn** [comment, marked as answer]
Solved! After 3, I need to directly flash the bin. No need to do step 4! 😭

**Reply by knmcguire** [reply]
ah yes, just adding some background info:  the making the app layer firmware is actually firmware + app in one. I you would do `tb make` you will be making just the firmware, but that will overwrite the first firmware + app.

Good you found the solution!
